clustering/ejb/WFLY-7628_Distributed_EJB_Timers.adoc

```diff
@@ -56,87 +59,108 @@ WildFly needs an persistent timer service implementation suitable for use in a c
 
 === Hard Requirements
 
-* Introduce a proper SPI for a TimerService within org.jboss.as.ejb3.timerservice.spi, to facilitate multiple implementations.
-** Adapt the existing TimerResource runtime attributes/operations to use the TimerService SPI instead of referencing specific implementation classes.
-** Adapt existing TimerService implementation to this SPI.
-* Add a CompositeTimerService implementation that delegates to separate persistent and transient TimerService implementations.
-* Implement a distributed implementation of the TimerService SPI.
-** Add a org.wildfly.clustering.ejb.timer SPI to the wildfly-clustering-ejb-spi module to facilitate multiple distributed timer manager implementations.
-** Add a distributed TimerService implementation to the EJB subsystem that delegates to the wildfly-clustering-ejb-spi for timer management and scheduling.
-** Add an Infinispan implementation of the org.wildfly.clustering.ejb.timer API that assigns execution ownership of timers to cluster members based on consistent hashing of the timer identifier.
-*** Leverage the existing Scheduler implementation within wildfly-clustering-ee-cache for timer scheduling
-*** Leverage the existing PrimaryOwnerScheduler decorator that uses consistent hashing of the timer ID to distribute timer execution among cluster members
-* Add a management resource and capability to the distributable-ejb subsystem to configure the Infinispan TimerManagement implementation.
-* Add an persistent-timer-management attribute to the EJB subsystem timer-service resource (as a alternate to the timer-store attribute) that references a TimerManagementProvider capability from the distributable-ejb subsystem.
-** Modify the TimerService DUP to use a CompositeTimerService in the event that the server is configured with a persistent-timer-management, that delegates to the existing TimerServiceImpl for use with transient timers and the DistributedTimerService for use with persistent timers.
+* Create a `TimerService` implementation that delegates to separate persistent vs transient `TimerService` implementations.
+* Implement Infinispan-based `TimerService` for persistent timers that manages timer execution based on the consistent hash of the timer identifier.
+** Make existing ``default-data-store`` attribute of the timer service resource optional.  Use distributable implementation when this attribute is undefined.
```

Comment:
This would mean that a new default server behavior would originate from this new feature in case it is configured to persist EJB timers and `default-data-store` is omitted, thus depending on the Infinispan subsystem configuration, right?

Reply:
When `default-data-store` is undefined, the default behavior for persistent timers will be determined via the distributable-ejb subsystem configuration.

Reply:
Ok, thanks, then it should be mentioned in the community docs.

Reply:
BTW @pferraro - Is the following:

```
Use distributable implementation when this attribute is undefined
```

tested somehow?

Reply:
https://github.com/pferraro/wildfly/blob/WFLY-7628/testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/timer/remote/HotRodPersistentTimerServiceTestCase.java#L68

Reply:
Thanks.